patsy/util.py

```diff
@@ -52,9 +52,21 @@
                                                "is_categorical_dtype", None)
 have_pandas_categorical_dtype = _pandas_is_categorical_dtype is not None
 
+if np.lib.NumpyVersion(np.__version__) >= "2.0.0":
```

Comment:
Can this simplified to only the try...except? With some comments.

Reply:
I have taken the changes to scipy as they are, but I can change them.
However, try-catch is generally considered to be expensive, which is why they use if-elif first in this way, I guess.

Reply:
+1. Let's just do the try/except. Thanks for stepping in here @bashtage !

Reply:
Thanks for your review!
I simplified it to just a try-except clause and added a comment.